fractional-NFT/project/fractional-NFT-contract/src/interface.sw

```diff
@@ -0,0 +1,57 @@
+library interface;
+
+dep data_structures;
+
+use data_structures::NFTInfo;
+
+abi FractionalNFT {
+    /// Locks an NFT into this contract and sends tokens to the caller.
+    ///
+    /// # Arguments
+    ///
+    /// * `admin` - The identity which will have the ability to withdraw.
+    /// * `nft` - The contract that manages the deposited NFT.
+    /// * `supply` - The number of fractionalized tokens that will be minted.
+    /// * `token_id` - The id of the NFT that is being deposited.
+    ///
+    /// # Reverts
+    ///
+    /// * When the contract has already been initialized.
```

Comment:
What does it mean to initialize a contract here without a constructor or an init function? I think it's better to explicitly mention that once a deposit is made then you cannot do it again or it will revert